ui/app/components/secret-edit.js

```diff
@@ -1,84 +1,63 @@
  * ```
-/
- * @param {object} model - Model returned from route secret-v2
+/This component is initialized from the secret-edit-layout.hbs file
+ * @param {object} model - Model returned from secret-v2 which is generated in the secret-edit route
+ * @param {string} mode - Edit, create, etc.
+ * @param {string} baseKey - Provided for navigation.
+ * @param {object} key - Passed through, copy of the model.
+ * @param {string} initialKey - model's name.
+ * @param {function} onRefresh - action that refreshes the model
+ * @param {function} onToggleAdvancedEdit - changes the preferAdvancedEdit to true or false
+ * @param {boolean} preferAdvancedEdit - property set from the controller of show/edit/create route passed in through secret-edit-layout
  */
 
 import { inject as service } from '@ember/service';
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { alias, or } from '@ember/object/computed';
-import FocusOnInsertMixin from 'vault/mixins/focus-on-insert';
-import WithNavToNearestAncestor from 'vault/mixins/with-nav-to-nearest-ancestor';
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 import KVObject from 'vault/lib/kv-object';
 import { maybeQueryRecord } from 'vault/macros/maybe-query-record';
+import { alias, or } from '@ember/object/computed';
 
-export default Component.extend(FocusOnInsertMixin, WithNavToNearestAncestor, {
-  wizard: service(),
-  store: service(),
-
-  // a key model
-  key: null,
-  model: null,
-
-  // a value to pre-fill the key input - this is populated by the corresponding
-  // 'initialKey' queryParam
-  initialKey: null,
-
-  // set in the route's setupController hook
-  mode: null,
-
-  secretData: null,
-
-  // called with a bool indicating if there's been a change in the secretData and customMetadata
-  onDataChange() {},
-  onRefresh() {},
-  onToggleAdvancedEdit() {},
-
-  // did user request advanced mode
-  preferAdvancedEdit: false,
-
-  // use a named action here so we don't have to pass one in
-  // this will bubble to the route
-  toggleAdvancedEdit: 'toggleAdvancedEdit',
-
-  codemirrorString: null,
+export default class SecretEdit extends Component {
+  @service wizard;
+  @service store;
 
-  isV2: false,
+  @tracked secretData = null;
+  @tracked isV2 = false;
+  @tracked codemirrorString = null;
 
-  init() {
-    this._super(...arguments);
-    let secrets = this.model.secretData;
-    if (!secrets && this.model.selectedVersion) {
-      this.set('isV2', true);
-      secrets = this.model.belongsTo('selectedVersion').value().secretData;
+  constructor() {
+    super(...arguments);
+    let secrets = this.args.model.secretData;
+    if (!secrets && this.args.model.selectedVersion) {
+      this.isV2 = true;
+      secrets = this.args.model.belongsTo('selectedVersion').value().secretData;
```

Comment:
This is a bit of an interesting way of accessing this property. You should be able to do `this.args.model.selectedVersion.secretData`.

Reply:
It's a copy over from how it was done before. I'm hesitant to change it in case I'm missing a reasoning for accessing the store again. maybe a refresh after choosing the version or something like that.